### PR TITLE
Implement generic Table components and DataTable

### PR DESCRIPTION
**Pull request by mathiasjakobsen**
`DataTable` can be used to align how we represent data in tables, and with a common way of overwriting components. This will be used in a dynamic search page, in AME

Screenshot 2022-08-31 at 15 02 14

### CODE REVIEW

**Review by Bartal — APPROVED** [review]

File: src/components/DataTable/index.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+                />
+              }
+            </CustomComponent>
+          )) }
+        </semanticComponents.Tr>
+      </semanticComponents.Thead>
+
+      <semanticComponents.Tbody>
+        { entries && entries.map((entry) => (
+          <semanticComponents.Tr childProps={entry}>
```
Comment (gvhuyssteen) [inline src/components/DataTable/index.js:70]: missing key

File: src/components/DataTable/index.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+            { !!indicators.length && 
+              <semanticComponents.Td className="flex items-center justify-between" childProps={entry}>
```
Comment (gvhuyssteen) [inline src/components/DataTable/index.js:72]: lowercase childprops

File: src/components/DataTable/index.js [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+                { indicators.map((CustomComponent) => (
+                  <CustomComponent {...entry} />
+                )) }
+              </semanticComponents.Td>
+            }
+
+            { Object.entries(bodyComponents).map(([key, CustomComponent]) => (
+              <semanticComponents.Td key={key} childProps={entry}>
```
Comment (gvhuyssteen) [inline src/components/DataTable/index.js:80]: lowercase childprops

File: src/components/DataTable/index.js (hunk 1 above)
Comment (gvhuyssteen) [inline src/components/DataTable/index.js:70]: lowercase childprops

File: src/components/Table/index.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+      'truncate',
+    ], {
+      'text-gray-800': highlighted,
+      'text-gray-600': !highlighted,
+    }, className)}
+    {...props}
+  />
+)
+
+const Table = ({ children, className = '' }) => (
```
Comment (gvhuyssteen) [inline src/components/Table/index.js:33]: className not being used

**Review by gvhuyssteen — CHANGES_REQUESTED** [review]

**Review by RynoVanHuyssteen77 — APPROVED** [review]

File: src/components/DataTable/index.js (hunk 2 above)
Comment (mathiasjakobsen) [reply src/components/DataTable/index.js:72]: Can you elaborate? All other props are `camelCase` in React?

Comment (gvhuyssteen) [reply src/components/DataTable/index.js:73]: Screenshot 2022-09-01 at 09 29 27

**Pull request merged**